physics/smoke_dust/dep_dry_mod_emerson.F90

```diff
@@ -0,0 +1,431 @@
+       real(kind_phys) :: Cc                      ! Cunningham/slip correction factor [-]
+       real(kind_phys) :: DDp, Eb                 ! Brownian diffusion []
+       real(kind_phys) :: Eim                     ! Impaction []
+       real(kind_phys) :: Ein                     ! Interception []
+       real(kind_phys) :: Sc                      ! Schmit number []
+       real(kind_phys) :: St                      ! Stokes number []
+       real(kind_phys) :: vg                      ! gravitational settling [cm/s]
+       real(kind_phys) :: A, eps0, dumalnsg       ! land surface params [-]
+       real(kind_phys) :: amu, amu_corrected      ! dynamic viscosity [g/s]
+       real(kind_phys) :: airkinvisc              ! Air kinetic viscosity [cm2/s]
+       real(kind_phys) :: freepath                ! Air molecular freepath [cm]
+       real(kind_phys) :: dp                      ! aerosol diameter [cm]
+       real(kind_phys) :: aerodens                ! aerosol density [g/cm3] 
+       real(kind_phys) :: Rs                      ! Surface resistance
+       real(kind_phys) :: vgpart
+       real(kind_phys) :: growth_fac,vsettl,dtmax,conver,converi,dzmin
+       real(kind_phys), dimension( kts:kte) :: rho_col, delz_col
+       real(kind_phys), dimension(ndvel)    :: dt_settl, chem_before, chem_after
+       real(kind_phys), dimension( kts:kte, ndvel ) :: cblk_col, vg_col
+       integer, dimension(ndvel) :: ndt_settl
+       integer :: i, j, k, ntdt, nv
+ ! chem pointers (p_*) are not sequentially numbered, need to define so nv loops work
+       integer, dimension(ndvel) :: chem_pointers
+!> -- Gas constant
+       real(kind_phys), parameter :: RSI = 8.314510
+       chem_pointers(1) = p_smoke
+       chem_pointers(2) = p_dust_1
+       chem_pointers(3) = p_coarse_pm
+  
+       growth_fac = 1.0
+       conver=1.e-9
+       converi=1.e9
+ 
+       do j = jts, jte
+          do i = its, ite
+             aer_res(i,j) = 0.0
+             do k = kts, kte
+                delz_col(k) = delz(i,k,j)
+                rho_col(k)  = rho_phy(i,k,j)
+                do nv = 1, ndvel
+                   cblk(nv) = chem(i,k,j,chem_pointers(nv))
+                   if ( k == kts ) then
+                      ddvel(i,j,nv) = 0.0
+                      dt_settl(nv)  = 0.0
+                   endif ! k==kts
+                end do ! nv
+                ! *** U.S. Standard Atmosphere 1962 page 14 expression
+                !     for dynamic viscosity = beta * T * sqrt(T) / ( T + S)
+                !     where beta = 1.458e-6 [ kg sec^-1 K**-0.5 ], s = 110.4 [ K ].
+                amu = 1.458E-6 * t_phy(i,k,j) * sqrt(t_phy(i,k,j)) / ( t_phy(i,k,j) + 110.4 )
+                ! Aerodynamic resistance
+                call depvel( rmol(i,j), dep_ref_hgt, znt(i,j), ustar(i,j), vgpart, aer_res(i,j) )
+                ! depvel uses meters, need to convert to s/cm
+                aer_res(i,j) = max(aer_res(i,j)/100.,0.) 
```

Comment:
For interoperability, change to explicit precision:
From aer_res(i,j) = max(aer_res(i,j)/100.,0.) 
To aer_res(i,j) = max(aer_res(i,j)/100._kind_phys,0._kind_phys)